### PR TITLE
BUG Fix page icons in vendor modules

### PR DESCRIPTION
**Pull request by tractorcow**
cc @axllent

### CODE REVIEW

**Comment by tractorcow** [comment]
Tests weren't working if cms installed as root, and there was a path case sensitivity issue. These are fixed now. :)

**Comment by tractorcow** [comment]
Ah, tests didn't work when cms was module root. Fixed now.

**Comment by tractorcow** [comment]
Fixed. :)

File: code/Controllers/LeftAndMainPageIconsExtension.php [hunk 1]
```diff
@@ -32,28 +35,25 @@ public function generatePageIconsCss()
...
+                $iconURL = $iconResource->getURL();
+                $css .= "$selector { background: transparent url('$iconURL') 0 0 no-repeat; }\n";
+            } elseif (Director::fileExists($icon)) {
+                // Support direct path
+                $iconURL = ModuleResourceLoader::resourceURL($icon);
+                $css .= "$selector { background: transparent url('$iconURL') 0 0 no-repeat; }\n";
             } else {
                 // Support for more sophisticated rules, e.g. sprited icons
-                $css .= "$selector { $iconFile }\n";
+                $css .= "$selector { $icon }\n";
```
Comment (flamerohr) [inline code/Controllers/LeftAndMainPageIconsExtension.php]: should this be sanitised?
I'm a bit wary of someone injecting css code here

Comment (tractorcow) [reply code/Controllers/LeftAndMainPageIconsExtension.php]: I think we may need this in Requirements_Backend

```diff
// Literal custom CSS content
foreach ($this->getCustomCSS() as $css) {
+    $css = Convert::raw2xml($css);
    $requirements .= HTML::createTag('style', ['type' => 'text/css'], "\n{$css}\n");
    $requirements .= "\n";
}
```

But separate story to this.

Comment (tractorcow) [reply code/Controllers/LeftAndMainPageIconsExtension.php]: Looks like it's not that easy.

Possible answer https://stackoverflow.com/questions/3241616/sanitize-user-defined-css-in-php, but requires a thirdparty lib.

Comment (tractorcow) [reply code/Controllers/LeftAndMainPageIconsExtension.php]: Let's dump this risky behaviour. It's a risk identified on owasp so let's at least minimise it https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet#HTML_entity_encoding

**Pull request merged**